psycopg3/transaction.py

```diff
@@ -0,0 +1,81 @@
+"""
+Transaction context managers returned by Connection.transaction()
+"""
+
+# Copyright (C) 2020 The Psycopg Team
+
+# TODO: Should this module be _underscored since it's implementation detail?
```

Comment:
Suggest moving `Transaction` class into `connection.py` because:
1. It's an implementation detail of `Connection`.
2. All the tests are in `test_connection.py` since that's where the functionality is exposed to the user.
3. It prevents circular import complications with type annotations.

Reply:
With type annotation there are plenty of circular imports. I guess you wrote this before discovering `if TYPE_CHECKING`?

Reply:
I am ok to move the tests into a module of its own. At the moment there is no `_module.py`, and other objects are details (transform, waiting...) Let's leave it this way for now, then we can make a rule about things.

Reply:
`transaction.py` and `test_transaction.py` sounds good. Will also shorten test names as they don't all need to start with `test_transaction`. +1